wordpress-mobile/WordPress-iOS
# Update Post Cell Statuses + Add Progress View

This PR extracts the logic for the status label in post list cells into a view model, which will make it easier to add uploading statuses for async. I've also updated the icons and colours for the statuses, to match our latest designs:

![4b15f725-7bb1-4452-bd2b-483814d295d7](https://user-images.githubusercontent.com/4780/36504255-ce791d02-1747-11e8-874e-c2addd0bb3f6.png)

Here are how these look in the cells:

![statuses](https://user-images.githubusercontent.com/4780/36504467-82527de6-1748-11e8-83be-91073a336e26.png)

Some of the states aren't in there yet as they're currently not supported. They'll be added as we add async support later.

I've also taken this opportunity to add a progress view to the bottom of the cell, which will be used when uploading posts / media, but it's currently hidden by default. 

**To test:**

* Check I've converted the logic correctly!
* If you want, you can either create posts of these different types (create a local post with no internet connection, schedule a post, mark a post as pending, trash a post), or edit the `postStatus` var in `PostCardStatusViewModel` to return different statuses.
* You can show the progress view by modifying `shouldHideProgressView` to return `false`
